daemon/src/send_async_safe.rs

```diff
@@ -0,0 +1,28 @@
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait SendAsyncSafe<M>
+where
+    M: xtra::Message<Result = ()>,
+{
+    /// Send a message to an actor without waiting for them to handle
+    /// it.
+    ///
+    /// As soon as this method returns, we know if the receiving actor
+    /// is alive. If they are, the message we are sending will
+    /// eventually be handled by them, but we don't wait for them to
+    /// do so.
+    async fn send_async_safe(&self, msg: M) -> Result<(), xtra::Disconnected>;
+}
+
+#[async_trait]
+impl<A, M> SendAsyncSafe<M> for xtra::Address<A>
```

Comment:
And implement this twice:

```rust
impl<A, M> SendAsyncSafe<M, ()> for xtra::Address<A> {}
impl<A, M, E: Error> SendAsyncSafe<M, Result<(), E>> for xtra::Address<A> {}
```
And in the 2nd impl we can do:

```rust
let fut = self.clone().send(msg);

tokio::spawn(async move {
	if let Err(e) = fut.await {
		tracing::warn!("Asynchronous message invocation failed: {:#}", e);
	}
})
```

Reply:
Two things:

- I cannot implement the same trait for the same generic type twice (even if the restrictions on its type parameter are different). Is there a workaround for this? I think we may have talked about this recently, but I can't recall.
- What is the difference between your second snippet and simply calling `do_send_async`?

Reply:
> I cannot implement the same trait for the same generic type twice (even if the restrictions on its type parameter are different). Is there a workaround for this? I think we may have talked about this recently, but I can't recall.

Not even if the trait is parameterized with different things? Note how the 2nd type parameter in `SendAsyncSafe` is different between the two impl blocks.

> What is the difference between your second snippet and simply calling do_send_async?

The 2nd impl will log an error if the async invocation fails whereas `do_send_async` will just drop the result, regardless of what it is.

Reply:
> Not even if the trait is parameterized with different things? Note how the 2nd type parameter in SendAsyncSafe is different between the two impl blocks.

I was doing something wrong and your tip fixed it!

> The 2nd impl will log an error if the async invocation fails whereas do_send_async will just drop the result, regardless of what it is.

Of course, that makes sense. Thanks!